vmware-base/src/com/cloud/hypervisor/vmware/mo/HypervisorHostHelper.java

```diff
@@ -501,12 +510,18 @@ public static void updatePortProfile(VmwareContext context, String ethPortProfil
             // No doubt about this, depending on vid=null to avoid lots of code below
             vid = null;
         } else {
+            if (vlanId != null) {
+                vlanId = vlanId.replace("vlan://", "");
+            }
             networkName = composeCloudNetworkName(namePrefix, vlanId, secondaryvlanId, networkRateMbps, physicalNetwork);
 
-            if (vlanId != null && !UNTAGGED_VLAN_NAME.equalsIgnoreCase(vlanId)) {
+            if (vlanId != null && !UNTAGGED_VLAN_NAME.equalsIgnoreCase(vlanId) && !vlanId.contains(",") && !vlanId.contains("-")) {
                 createGCTag = true;
                 vid = Integer.parseInt(vlanId);
             }
+            if (vlanId != null && (vlanId.contains(",") || vlanId.contains("-"))) {
```

Comment:
could be replaced by  StringUtils.containsAny(vlanId, ",-")? Maybe also extract it to a separate boolean because it's checked multiple times?

Reply:
Fixed, thanks.